initial search on auctions data. 

In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)
getty2 = pd.read_csv("auctions.csv", sep=";", encoding="iso-8859-1")

getty2

Column1             artist   country yearOfBirth  yearOfDeath  \
0          0.0            Mario A     Swiss        1959          NaN   
1          1.0            Mario A     Swiss        1959          NaN   
2          2.0         A E Cremer    French       20thC          NaN   
3          3.0  A G Schultz & Co.  American    est1890s          NaN   
4          4.0  A G Schultz & Co.  American    est1890s          NaN   
...        ...                ...       ...         ...          ...   
44097  41290.0        Mark Rothko  American      1903.0      19700.0   
44098  41291.0        Mark Rothko  American      1903.0      19700.0   
44099  41292.0        Mark Rothko  American      1903.0      19700.0   
44100  41293.0        Mark Rothko  American      1903.0      19700.0   
44101  41294.0        Mark Rothko  American      1903.0      19700.0   

                                                    name  year       price  \
0                              The world is beautyful #5  2004     53150.0   
1                             The world is beautyful #13  2004     73830.0   
2                                            Spot Lights   NaN     20900.0   
3                                Sugar/Sweetmeat Baskets   NaN      6150.0   
4      An American Silver Five-piece Tea And Coffee S...  1908     81250.0   
...                                                  ...   ...         ...   
44097                                           Untitled  1961  17000000.0   
44098                                           Untitled  1968   3000000.0   
44099                                           Untitled  1959   2800000.0   
44100                        Green, White, Green on Blue  1969   3229380.0   
44101                                         Red on red  1968   3300000.0   

                            material height  width  \
0           laserchrome_print_diasec  29.53  39.37   
1         laserchrom_print_(diasec.)  29.53  39.37   
2                black-painted_metal    NaN    NaN   
3                    Sterling_Silver    6.5   5.75   
4                             silver    NaN    NaN   
...                              ...    ...    ...   
44097                  oil_on_canvas   93.0   80.0   
44098     acrylic_on_paper_on_canvas   24.0   18.0   
44099  oil_on_paper_laid_on_masonite   38.0   25.0   
44100          oil_on_paper_on_board   71.0   41.0   
44101         oil_on_paper_on_canvas   40.0   25.0   

                                                    link  \
0      http://artsalesindex.artinfo.com/asi/lots/3456055   
1      http://artsalesindex.artinfo.com/asi/lots/3456059   
2      http://artsalesindex.artinfo.com/asi/lots/5250074   
3      http://artsalesindex.artinfo.com/asi/lots/5269624   
4      http://artsalesindex.artinfo.com/asi/lots/3900522   
...                                                  ...   
44097  http://artsalesindex.artinfo.com/asi/lots/2495060   
44098  http://artsalesindex.artinfo.com/asi/lots/2495081   
44099  http://artsalesindex.artinfo.com/asi/lots/2495098   
44100  http://artsalesindex.artinfo.com/asi/lots/2251703   
44101  http://artsalesindex.artinfo.com/asi/lots/2085679   

                                                  source dominantColor  \
0      http://artinfo-images-350.s3.amazonaws.com/asi...       yellows   
1      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
2      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
3      http://artinfo-images-350.s3.amazonaws.com/asi...        blacks   
4      http://artinfo-images-350.s3.amazonaws.com/asi...        whites   
...                                                  ...           ...   
44097  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44098  http://artinfo-images-350.s3.amazonaws.com/177...          reds   
44099  http://artinfo-images-350.s3.amazonaws.com/177...        blacks   
44100  http://artinfo-images-350.s3.amazonaws.com/S11...        blacks   
44101  http://artinfo

In [2]:
from collections import Counter
liss = {}
for column_name in getty2.columns:
    if column_name == "artist":
        for value in getty2[column_name]:
            if value not in liss:
                liss[value] = 0
            else:
                liss[value] += 1
k = Counter(liss)
 
# Finding 3 highest values
high = k.most_common(4) 
 
print("Initial Dictionary:")
print(liss, "\n")
 
 
print("Dictionary with 3 highest values:")
print("Keys: Values")
 
for i in high:
    print(i[0]," :",i[1]," ")

Initial Dictionary:
{'Mario A': 1, 'A E Cremer': 0, 'A G Schultz & Co.': 1, 'A H': 0, 'A H G': 0, 'A K': 0, 'A Kassen': 0, 'A Lao': 3, 'A M': 0, 'A Sun Wu': 3, "Maria A'Becket": 3, "A'erbai": 0, 'A-One': 1, 'A.CE': 0, 'A1one': 5, 'Dirk van der Aa': 5, 'Jorgen Aabye': 4, 'Christian Aabye-Talge': 0, 'Attributed to  Hans von Aachen': 1, 'Circle of  Hans von Aachen': 1, 'Studio of  Hans von Aachen': 0, 'After  Hans von Aachen': 0, 'Hans von Aachen': 0, 'Peder Pedersen Aadnes': 0, 'Arvid Aae': 1, 'Carl Frederic Aagaard': 3, 'Martin Aagaard': 1, 'H P Aaland': 0, 'Alvar Aalto': 10, nan: 2836, 'Ilmari Aalto': 3, 'Veikko Aaltona': 4, 'Aarre Aaltonen': 1, 'Waino Aaltonen': 6, 'Martin Aalund': 2, 'Asbjorn Aamodt': 1, 'Susanne Aamund': 0, 'Jos Aanraad': 2, 'Leny Aardse': 0, 'Eero Aarnio': 6, 'Jessie Aaron': 8, 'John Aaron': 0, 'Joseph Aaron': 4, 'Michel Aaron': 0, 'George Aarons': 3, 'Slim Aarons': 1, 'Carl Aarsleff': 1, 'Hans Aarsman': 0, 'Nicolaas Aartmann': 3, 'Pieter Jan Aarts': 0, 'Alf Jorgen

In [3]:
liss = []
for column_name in getty2.columns:
    if column_name == "artist":
        for value in getty2[column_name]:
            if value == "Gustav Klimt":
                liss.append(value)
print(len(liss))

424


Get only results related to sales on Klimt works. Artworks that do not possess data about price or soldtime have been deleted from the dataset.
The dataset does not possess any data related to sellers and buyers. 
In this example we notice that 256 works over a total of 424 possess data about price and time of the transaction. 

In [4]:
import pandas as pd

new_dataframe = getty2[["artist", "name", "price", "soldtime"]].copy()
new_dataframe = new_dataframe[new_dataframe["artist"] == "Gustav Klimt"].copy()

new_dataframe.reset_index(drop=True, inplace=True)

for idx, row in new_dataframe.iterrows():
    if pd.isna(row["price"]) and pd.isna(row["soldtime"]):
        new_dataframe.drop(idx, inplace=True)
        
dayy = []
for i in new_dataframe['soldtime']:
    string_representation = str(i)
    split_parts = string_representation.split('-')
    day = split_parts[0]
    dayy.append(day)
new_dataframe['soldtime'] = dayy
        

new_dataframe.reset_index(drop=True, inplace=True)
new_dataframe

artist                                               name  \
0    Gustav Klimt                       Vorgebeugt sitzende Aktfigur   
1    Gustav Klimt  Stehender Frauenakt Mit Erhobenem Linkem Bein,...   
2    Gustav Klimt                       Nach Links Sitzender Halbakt   
3    Gustav Klimt                            Allegory of Agriculture   
4    Gustav Klimt  Mädchenbrustbild Von Vorne Mit Häubchen (Por...   
..            ...                                                ...   
251  Gustav Klimt                    Young nude with long, open hair   
252  Gustav Klimt  Head and shoulders portrait of a lady glancing...   
253  Gustav Klimt                              Female nude bent over   
254  Gustav Klimt                                             Lovers   
255  Gustav Klimt            Seated female nude, turned to the right   

         price soldtime  
0     174600.0     2014  
1     225000.0     2014  
2     200000.0     2014  
3     200000.0     2014  
4    2691170.0     2014  
..         ...      ...  
251   288000.0     2004  
252   120140.0     2004  
253   192220.0     2004  
254   180210.0     2004  
255   168190.0     2004  

[256 rows x 4 columns]